### DISCUSSION TITLE
Github Apps to add secrets

### DISCUSSION THREAD

**Discussion by wei** [community/community#25270] [Actions]
<p>Can Github Apps API add secrets to repos to be used in Github Actions? It doesn’t appear to be <a href="https://developer.github.com/v3/apps/permissions/" rel="nofollow noopener">listed</a> but would be tremendously helpful. </p>
<p>It is also not yet in the <a href="https://developer.github.com/v3/repos/" rel="nofollow noopener">REST v3 Repositories API</a> so even users can’t do it with their personal access tokens. </p>
<p>Is this in the timeline soon and am I missing anything? Thanks so much in advance!</p>

**Comment by lee-dohm** [comment]
<p>No, it isn’t currently possible to add secrets using either the REST or GraphQL API. I can’t make any promises as to when or if it will be added, but I’ll pass along the feedback to the team.</p>
<p>Thanks for reaching out.</p>

**Comment by joe-bowman** [comment]
<p>+1 on this. We want to be able to update the secret with short-TTL temporary credentials to limit exposure due to unintended disclosure. The ability to automatically update / rotate secrets is a must.</p>

**Comment by jikuma** [comment]
<p>+1 for this. We require this capability to automate secret rotation. </p>

**Comment by davidanthoff** [comment]
<p>Same here, having an API that allows us to add secrets would be really key for us.</p>

**Comment by zhuzeng** [comment]
<p>We definitely need this, without this we cannot use Github Actions (we have workflows that we MUST dynamically generate the secrets and upload to Github, we cannot do this operation manually through UI). Is there a timeline or workaround?</p>

**Comment by ajcann** [comment]
<p>Any update on this? Pretty surprising the API doesn’t have endpoints for this.</p>

**Comment by mscoutermarsh** [comment]
<p>Hi all,</p>
<p>We’re working on building this for you all now. I don’t have a date yet for when it will ship. Will announce on the change log when available.</p>

**Comment by okor** [comment]
<p>Thank you <a class="mention" href="/mscoutermarsh">@mscoutermarsh</a>!</p>

**Comment by andymed-jlp** [comment]
<p>Thanks! That will be really useful for us.</p>

**Comment by in8sudku** [comment]
<p>any update on this?</p>

**Comment by amadosdiop** [comment]
<p>+1. </p>
<p>It will be really useful</p>

**Comment by mscoutermarsh** [comment]
<p>Soon :).</p>
<p>Keep an eye on <a href="https://twitter.com/ghchangelog" rel="nofollow noopener">https://twitter.com/ghchangelog</a>.</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Update: this is now available :slight_smile:</p>
<p><a href="https://developer.github.com/v3/actions/secrets/" rel="nofollow noopener">https://developer.github.com/v3/actions/secrets/</a></p>

**Comment by mpdude** [comment]
<p>We have written a tool to track secrets usage and provide a way for centralized rotation: <a href="https://github.com/webfactory/secret-spreader" rel="nofollow noopener">https://github.com/webfactory/secret-spreader</a></p>
<p>It is based on the brand-new (beta) GitHub API for Actions. See the README over at the repo for full details.</p>
<p>Feedback is very much appreciated!</p>

**Comment by blacha** [comment]
<p>Awesome work!</p>
<p>With this API I have written a CLI tool, to quickly rotate all my github action’s AWS credentials.</p>
<p><a href="https://github.com/blacha/github-secret-rotate" rel="nofollow noopener">https://github.com/blacha/github-secret-rotate</a></p>

**Comment by kasthuriraajan** [comment]
<p>I couldn’t find this feature in Github GraphQL API. Does this feature available in GraphQL API too ?</p>
